backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/gson/spi/GsonProvider.java

```diff
@@ -4,7 +4,7 @@
  * License: GNU Lesser General Public License (LGPL), version 2.1 or later
  * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
  */
-package org.hibernate.search.backend.elasticsearch.gson.impl;
+package org.hibernate.search.backend.elasticsearch.gson.spi;
```

Comment:
Do we really want to expose Gson in our SPI?

Reply:
We have to, if we want to allow to replace the Elasticsearch client: it takes JSON as input. We could format it to a String, but that would prevent some optimizations in the client itself (related to non-blocking requests).

I would not expose Gson in APIs, but in SPIs, especially this single SPI related to the Elasticsearch client, I think it's safe. If we were to replace Gson with something else, we wouldn't do it in a micro anyway, would we? And we allow ourselves to break SPIs in minors.